sqlmesh/magics.py

```diff
@@ -60,8 +61,14 @@ def wrapper(self: SQLMeshMagics, *args: t.Any, **kwargs: t.Any) -> None:
         if bound_method:
             args_split = arg_split(args[0])
             parser = bound_method.parser
-            # Calling the private method to bypass setting of defaults.
-            parsed_args, _ = parser._parse_known_args(args_split, Namespace())
+
+            # Temporarily supress default values, otherwise any missing arg would be set and affect analytics
+            original_parser_actions = deepcopy(parser._actions)
```

Comment:
Do we also need to clear `_defaults`?

Reply:
I remember peeking at `_defaults` and it was empty, I'll revisit and cross-check with the current implementation.

Reply:
I cleared it to be safe, since there seems to be a [code path](https://github.com/python/cpython/blob/main/Lib/argparse.py#L1942-L1945) where the namespace can be mutated if defaults are present.